test/rubygems/test_gem_commands_uninstall_command.rb

```diff
@@ -78,7 +79,7 @@ def test_execute_dependency_order
   def test_execute_removes_executable
     ui = Gem::MockGemUi.new
 
-    util_setup_gem ui
+    @installer = util_setup_gem ui
```

Comment:
Why we need to assign to an instance variable?

Reply:
No particular reason.

I migrated everything from the `setup` method to be defined locally to each test method but kept the instance variables because the helper methods still defined some instance variables that were still referenced in the tests. But the `@installer` variable in particular can be safely changed to be a local variable. I can do that in a follow up commit :+1:.

Reply:
I addressed your feedback, and also fixed a state thas was surfaced by this refactoring in  https://github.com/rubygems/rubygems/pull/2757/commits/aa14595f38f2f3424d2960db4463aedbba6f2e78.